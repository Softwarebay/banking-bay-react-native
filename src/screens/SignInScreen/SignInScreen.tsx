import { useNavigation } from '@react-navigation/native';

import type { NavigationProp } from 'navigator';

import { SignInTemplate } from 'components/templates';

export const SignInScreen = () => {
	const navigation = useNavigation<NavigationProp>();

	const signInViaEmailPassword = (userCredentials: {
		email: string;
		password: string;
		persistPassword: boolean;
	}) => {
		console.log(userCredentials);
		navigation.navigate('Root');
	};

	const restorePassword = () => {
		console.log('Restore password');
	};

	const navigateToSignUp = () => {
		console.log('To SignUp');
	};

	const signInViaFacebook = () => {
		console.log('Sign in via Facebook');
	};

	const signInViaGooglePlus = () => {
		console.log('Sign in via Google+');
	};

	const signInViaTwitter = () => {
		console.log('Sign in via Twitter');
	};

	return (
		<SignInTemplate
			footerLinkText="Register now"
			footerText="No account?"
			headingText="Welcome to BankingBay!"
			onPressFacebook={signInViaFacebook}
			onPressFooterLink={navigateToSignUp}
			onPressGooglePlus={signInViaGooglePlus}
			onPressTwitter={signInViaTwitter}
			onRestorePassword={restorePassword}
			onSubmitSignInForm={signInViaEmailPassword}
		/>
	);
};
