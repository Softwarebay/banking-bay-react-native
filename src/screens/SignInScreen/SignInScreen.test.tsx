import { fireEvent, render, screen } from '@testing-library/react-native';
import testRenderer from 'react-test-renderer';

import { SignInScreen } from './SignInScreen';

const mockNavigate = jest.fn();
jest.mock('@react-navigation/native', () => ({
	useNavigation: () => ({ navigate: mockNavigate }),
}));

describe('SignInScreen', () => {
	it('should match snapshot', () => {
		const tree = testRenderer.create(<SignInScreen />).toJSON();

		expect(tree).toMatchSnapshot();
	});

	it("should display 'Welcome to BankingBay!' as screen heading", () => {
		render(<SignInScreen />);

		const heading = screen.getByTestId('sign-in-heading-test-id');

		expect(heading).toHaveTextContent('Welcome to BankingBay!');
	});

	it("should display 'No account?' as form footer text", () => {
		render(<SignInScreen />);

		const footerText = screen.getByTestId('sign-in-footer-text-test-id');

		expect(footerText).toHaveTextContent('No account?');
	});

	it("should display 'Register now' as form footer link", () => {
		render(<SignInScreen />);

		const footerLink = screen.getByTestId('sign-in-footer-link-test-id');

		expect(footerLink).toHaveTextContent('Register now');
	});

	it("should log 'Restore password' after pressing on restore password link", () => {
		render(<SignInScreen />);
		console.log = jest.fn();

		const restorePassLink = screen.getByTestId(
			'sign-in-form-restore-password-link-test-id'
		);

		fireEvent.press(restorePassLink);

		expect(console.log).toHaveBeenCalledWith('Restore password');
	});

	it("should log 'To SignUp' after pressing on form footer link", () => {
		render(<SignInScreen />);
		console.log = jest.fn();

		const footerLink = screen.getByTestId('sign-in-footer-link-test-id');

		fireEvent.press(footerLink);

		expect(console.log).toHaveBeenCalledWith('To SignUp');
	});

	it('should log user credentials after successful sign in', () => {
		render(<SignInScreen />);
		console.log = jest.fn();

		const submitBtn = screen.getByTestId('sign-in-form-submit-button-test-id');

		fireEvent.press(submitBtn);

		expect(console.log).toHaveBeenCalledWith({
			email: 'test@test.com',
			password: 'password',
			persistPassword: false,
		});
	});

	it('should navigate to DashboardScreen after successful sign in', () => {
		render(<SignInScreen />);

		const submitBtn = screen.getByTestId('sign-in-form-submit-button-test-id');

		fireEvent.press(submitBtn);

		expect(mockNavigate).toHaveBeenCalledWith('Root');
	});

	it("should log 'Sign in via Facebook' after pressing on Facebook icon button", () => {
		render(<SignInScreen />);
		console.log = jest.fn();

		const facebookIconBtn = screen.getByTestId(/facebook-icon-btn-test-id/i);

		fireEvent.press(facebookIconBtn);

		expect(console.log).toHaveBeenCalledWith('Sign in via Facebook');
	});

	it("should log 'Sign in via Google+' after pressing on Google+ icon button", () => {
		render(<SignInScreen />);
		console.log = jest.fn();

		const googlePlusIconBtn = screen.getByTestId(
			/googlePlus-icon-btn-test-id/i
		);

		fireEvent.press(googlePlusIconBtn);

		expect(console.log).toHaveBeenCalledWith('Sign in via Google+');
	});

	it("should log 'Sign in via Twitter' after pressing on Twitter icon button", () => {
		render(<SignInScreen />);
		console.log = jest.fn();

		const twitterIconBtn = screen.getByTestId(/twitter-icon-btn-test-id/i);

		fireEvent.press(twitterIconBtn);

		expect(console.log).toHaveBeenCalledWith('Sign in via Twitter');
	});
});
