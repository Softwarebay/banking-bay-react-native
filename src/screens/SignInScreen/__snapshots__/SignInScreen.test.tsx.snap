// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInScreen should match snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FAFCFF",
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 20,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#1B1D4D",
        },
        Object {
          "fontFamily": "MulishBold",
          "fontSize": 51,
          "lineHeight": 56.1,
        },
        Object {
          "marginBottom": 30,
          "paddingHorizontal": 35,
          "textAlign": "center",
        },
      ]
    }
    testID="sign-in-heading-test-id"
  >
    Welcome to BankingBay!
  </Text>
  <View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
          },
          Object {
            "flexDirection": "row",
          },
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 10,
              "shadowColor": "#161966",
              "shadowOffset": Object {
                "height": 6,
                "width": 0,
              },
              "shadowOpacity": 0.08,
              "shadowRadius": 16,
            },
            Array [
              Object {
                "height": 55,
                "paddingHorizontal": 20,
                "paddingVertical": 14,
              },
              Object {
                "marginBottom": 14,
              },
            ],
          ],
        ]
      }
      testID="sign-in-form-email-input-test-id"
    >
      <TextInput
        autoCorrect={false}
        keyboardType="email-address"
        onChangeText={[Function]}
        placeholder="Enter your email"
        secureTextEntry={false}
        style={
          Array [
            Object {
              "color": "#1B1D4D",
              "fontFamily": "MulishRegular",
              "fontSize": 15,
            },
            Object {
              "flex": 1,
              "marginRight": 20,
            },
          ]
        }
        testID="input-field-sign-in-form-email-input-test-id"
        value=""
      />
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
          },
          Object {
            "flexDirection": "row",
          },
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 10,
              "shadowColor": "#161966",
              "shadowOffset": Object {
                "height": 6,
                "width": 0,
              },
              "shadowOpacity": 0.08,
              "shadowRadius": 16,
            },
            Object {
              "height": 55,
              "paddingHorizontal": 20,
              "paddingVertical": 14,
            },
          ],
        ]
      }
      testID="sign-in-form-password-input-test-id"
    >
      <TextInput
        autoCorrect={false}
        keyboardType="default"
        onChangeText={[Function]}
        placeholder="Enter your password"
        secureTextEntry={true}
        style={
          Array [
            Object {
              "color": "#1B1D4D",
              "fontFamily": "MulishRegular",
              "fontSize": 15,
            },
            Object {
              "flex": 1,
              "marginRight": 20,
            },
          ]
        }
        testID="input-field-sign-in-form-password-input-test-id"
        value=""
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        hitSlop={20}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "right": 0,
            },
            false,
          ]
        }
        testID="icon-btn-sign-in-form-password-input-test-id"
      >
        <SvgMock
          color="#1B1D4D"
          height={16}
          testID="eye-off-icon-test-id"
          width={16}
        />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
          },
          Object {
            "flexDirection": "row",
          },
          Object {
            "justifyContent": "space-between",
            "marginBottom": 30,
            "marginTop": 20,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
            },
            Object {
              "flexDirection": "row",
            },
            undefined,
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          hitSlop={20}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderColor": "#868698",
              "borderRadius": 4,
              "borderWidth": 1,
              "height": 22,
              "justifyContent": "center",
              "width": 22,
            }
          }
          testID="sign-in-form-checkbox-test-id"
        />
        <Text
          style={
            Array [
              Object {
                "color": "#4C4C60",
                "fontFamily": "MulishRegular",
                "fontSize": 17,
                "lineHeight": 27.2,
              },
              Object {
                "marginLeft": 14,
              },
            ]
          }
        >
          Remember me
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        hitSlop={20}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={false}
        testID="sign-in-form-restore-password-link-test-id"
      >
        <Text
          style={
            Array [
              Object {
                "color": "#FF5887",
                "fontFamily": "MulishMedium",
                "fontSize": 17,
                "lineHeight": 27.2,
              },
              undefined,
            ]
          }
        >
          Lost your password?
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          false,
          undefined,
        ]
      }
      testID="sign-in-form-submit-button-test-id"
    >
      <ViewManagerAdapter_ExpoLinearGradient
        proxiedProperties={
          Object {
            "colors": Array [
              4288142079,
              4279654301,
            ],
            "endPoint": Array [
              1,
              1,
            ],
            "locations": undefined,
            "startPoint": Array [
              0,
              0,
            ],
          }
        }
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 10,
            "height": 55,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontFamily": "MulishSemiBold",
                "fontSize": 17,
              },
              Object {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          Sign In
        </Text>
      </ViewManagerAdapter_ExpoLinearGradient>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
        },
        Object {
          "flexDirection": "row",
        },
        Object {
          "marginBottom": 50,
          "marginTop": 30,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#4C4C60",
            "fontFamily": "MulishRegular",
            "fontSize": 17,
            "lineHeight": 27.2,
          },
          undefined,
        ]
      }
      testID="sign-in-footer-text-test-id"
    >
      No account? 
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={20}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={false}
      testID="sign-in-footer-link-test-id"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FF5887",
              "fontFamily": "MulishMedium",
              "fontSize": 17,
              "lineHeight": 27.2,
            },
            undefined,
          ]
        }
      >
        Register now
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
        },
        Object {
          "flexDirection": "row",
        },
        Object {
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={20}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
          Array [
            Object {
              "borderRadius": 100,
              "height": 60,
              "marginLeft": 10,
              "width": 60,
            },
            Object {
              "backgroundColor": "#3B5999",
            },
          ],
          false,
        ]
      }
      testID="Facebook-icon-btn-test-id"
    >
      <SvgMock
        color="#FFFFFF"
        height={27}
        testID="facebook-icon-test-id"
        width={27}
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={20}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
          Array [
            Object {
              "borderRadius": 100,
              "height": 60,
              "marginLeft": 10,
              "width": 60,
            },
            Object {
              "backgroundColor": "#55ACEE",
            },
          ],
          false,
        ]
      }
      testID="Twitter-icon-btn-test-id"
    >
      <SvgMock
        color="#FFFFFF"
        height={27}
        testID="twitter-icon-test-id"
        width={27}
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={20}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
          Array [
            Object {
              "borderRadius": 100,
              "height": 60,
              "marginLeft": 10,
              "width": 60,
            },
            Object {
              "backgroundColor": "#DD4B39",
            },
          ],
          false,
        ]
      }
      testID="GooglePlus-icon-btn-test-id"
    >
      <SvgMock
        color="#FFFFFF"
        height={27}
        testID="google-plus-icon-test-id"
        width={27}
      />
    </View>
  </View>
</View>
`;
